Model/PayolutionInvoicePaymentMethod.php

```diff
@@ -107,8 +129,95 @@ public function getConfigFields()
                 'title' => Helper::translate('wd_config_delete_failure_order'),
                 'description' => Helper::translate('wd_config_delete_failure_order_desc'),
             ],
+            'allowedCurrencies' => [
+                'type' => 'multiselect',
+                'field' => 'oxpayments__allowed_currencies',
+                'options' => PaymentMethodHelper::getCurrencyOptions(),
+                'title' => Helper::translate('wd_config_allowed_currencies'),
+                'description' => Helper::translate('wd_config_allowed_currencies_desc'),
+                'required' => true,
+            ],
+            'apiUrl' => [
+                'type' => 'text',
+                'field' => 'oxpayments__wdoxidee_apiurl',
+                'title' => Helper::translate('wd_config_base_url'),
+                'description' => Helper::translate('wd_config_base_url_desc'),
+                'required' => true,
+            ],
         ];
-        return parent::getConfigFields() + $aAdditionalFields;
+        return $aConfigFields + $this->_getCustomCurrencyConfigFields();
+    }
+
+    /**
+     * Returns an array of all custom config fields per activated currency.
+     * One separator, HTTP user/password and MAID and secret fields are added per currency.
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    private function _getCustomCurrencyConfigFields()
+    {
+        $aCurrencyFields = [];
+
+        $aCurrencies = $this->_oPayment->oxpayments__allowed_currencies->value;
+
+        // fields that are configurable per currency
+        $aFields = [
+            'httpUser' => [
+                'fieldType' => 'text',
```

Comment:
(Minor) I would just call this `type` and `field` below to stick with the same syntax that we have for regular fields (the other keys are also the same).

Reply:
👍